modules/runner/src/integrationTest/java/org/apache/ignite/internal/benchmark/InsertBenchmark.java

```diff
@@ -0,0 +1,203 @@
+import org.openjdk.jmh.annotations.Measurement;
+import org.openjdk.jmh.annotations.Mode;
+import org.openjdk.jmh.annotations.OutputTimeUnit;
+import org.openjdk.jmh.annotations.Scope;
+import org.openjdk.jmh.annotations.Setup;
+import org.openjdk.jmh.annotations.State;
+import org.openjdk.jmh.annotations.TearDown;
+import org.openjdk.jmh.annotations.Warmup;
+import org.openjdk.jmh.runner.Runner;
+import org.openjdk.jmh.runner.RunnerException;
+import org.openjdk.jmh.runner.options.Options;
+import org.openjdk.jmh.runner.options.OptionsBuilder;
+
+/**
+ * Benchmark for insertion operation, comparing KV, JDBC and SQL APIs.
+ */
+@State(Scope.Benchmark)
+@OutputTimeUnit(TimeUnit.MICROSECONDS)
+public class InsertBenchmark extends AbstractOneNodeBenchmark {
+    /**
+     * Benchmark for SQL insert.
+     */
+    @Benchmark
+    @Warmup(iterations = 1, time = 10)
+    @Measurement(iterations = 1, time = 20)
+    public void sqlInsert(SqlState state) {
+        state.executeQuery();
+    }
+
+    /**
+     * Benchmark for KV insert.
+     */
+    @Benchmark
+    @Warmup(iterations = 1, time = 10)
+    @Measurement(iterations = 1, time = 20)
+    public void kvInsert(KvState state) {
+        state.executeQuery();
+    }
+
+    /**
+     * Benchmark for JDBC insert.
+     */
+    @Benchmark
+    @Warmup(iterations = 1, time = 10)
+    @Measurement(iterations = 1, time = 20)
+    public void jdbcInsert(JdbcState state) throws SQLException {
+        state.executeQuery();
+    }
+
+    /**
+     * Benchmark's entry point.
+     */
+    public static void main(String[] args) throws RunnerException {
+        Options opt = new OptionsBuilder()
+                .include(".*" + InsertBenchmark.class.getSimpleName() + ".*")
+                .forks(0)
+                .threads(1)
+                .mode(Mode.SampleTime)
+                .build();
+
+        new Runner(opt).run();
+    }
+
+    /**
+     * Benchmark state for {@link #sqlInsert(SqlState)}. Holds {@link Session} and {@link Statement}.
+     */
+    @State(Scope.Benchmark)
+    public static class SqlState {
+        private Statement statement;
+        private Session session;
+
+        /**
+         * Initializes session and statement.
+         */
+        @Setup
+        public void setUp() {
+            String fieldsQ = IntStream.range(1, 11).mapToObj(i -> "field" + i).collect(joining(","));
+            String valQ = IntStream.range(1, 11).mapToObj(i -> "'" + FIELD_VAL + "'").collect(joining(","));
+
+            String queryStr = String.format("insert into %s(%s, %s) values(?, %s);", TABLE_NAME, "ycsb_key", fieldsQ, valQ);
+
+            IgniteSql sql = clusterNode.sql();
+
+            statement = sql.createStatement(queryStr);
+            session = sql.createSession();
+        }
+
+        /**
+         * Closes resources.
+         */
+        @TearDown
+        public void tearDow() throws Exception {
+            IgniteUtils.closeAll(session/*, statement*/);
```

Comment:
Why statement is commented?

Reply:
Because it throws UnsupportedOperationException, I need to add a comment